### PR TITLE
[fix](segment cache) estimate momory consumed by segment

### PR DESCRIPTION
**Pull request by dataroaring**
The memory consumed in segment cache is 0 after
https://github.com/apache/doris/pull/35432/files.

The pr also tracks memory usage of column readers.

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run performance

**Review by Lchangliang — APPROVED** [review]

**Comment by Lchangliang** [comment]
LGTM

File: be/src/olap/rowset/rowset.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 namespace doris {
 
+static bvar::Adder<size_t> g_total_rowset_num("doris_total_rowset_num");
```
Comment (xinyiZzz) [inline be/src/olap/rowset/rowset.cpp:29]: BE metrics have a rowset_num counter, is it same as bvar::doris_total_rowset_num?
`update_rowsets_and_segments_num_metrics`
image

File: be/src/olap/rowset/segment_v2/ordinal_page_index.cpp [hunk 2]
```diff
@@ -29,9 +29,13 @@
 #include "olap/olap_common.h"
 #include "olap/rowset/segment_v2/page_handle.h"
 #include "olap/rowset/segment_v2/page_io.h"
+#include "ordinal_page_index.h"
 #include "util/slice.h"
 
 namespace doris {
+
+static bvar::Adder<size_t> g_ordinal_index_memory_bytes("doris_ordinal_index_memory_bytes");
```
Comment (xinyiZzz) [inline be/src/olap/rowset/segment_v2/ordinal_page_index.cpp:37]: should create a global MemTracker for this counter

File: be/src/olap/rowset/segment_v2/indexed_column_reader.cpp [hunk 3]
```diff
@@ -56,6 +56,8 @@ static bvar::Adder<uint64_t> g_index_reader_pk_pages("doris_pk", "index_reader_p
 static bvar::PerSecond<bvar::Adder<uint64_t>> g_index_reader_pk_bytes_per_second(
         "doris_pk", "index_reader_pk_pages_per_second", &g_index_reader_pk_pages, 60);
 
+static bvar::Adder<uint64_t> g_index_reader_memory_bytes("doris_index_reader_memory_bytes");
```
Comment (xinyiZzz) [inline be/src/olap/rowset/segment_v2/indexed_column_reader.cpp:59]: should create a global MemTracker for this counter

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 4]
```diff
@@ -73,7 +73,7 @@
 #include "vec/olap/vgeneric_iterators.h"
 
 namespace doris::segment_v2 {
-bvar::Adder<size_t> g_total_segment_num("doris_total_segment_num");
+static bvar::Adder<size_t> g_total_segment_num("doris_total_segment_num");
```
Comment (xinyiZzz) [inline be/src/olap/rowset/segment_v2/segment.cpp:76]: BE metrics have a counter `all_segments_num`, is it same as bvar::doris_total_segment_num?

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Comment by xinyiZzz** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Pull request merged**